src/EditorFeatures/Core/Implementation/ForegroundNotification/ForegroundNotificationService.cs

```diff
@@ -139,6 +147,15 @@ private void NotifyOnForegroundWorker()
                         {
                             // eat up cancellation
```

Comment:
hmmm I am not sure why it is matter whether this runs from BG or not. InvokeBelowInputPriority should make sure given action run on UI thread.